src/elevateprivilegeshelper.h

```diff
@@ -0,0 +1,76 @@
+/*
+    SPDX-FileCopyrightText: Zeno Endemann <zeno.endemann@kdab.com>
+    SPDX-FileCopyrightText: 2016-2023 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+
+    SPDX-License-Identifier: GPL-2.0-or-later
+*/
+
+#pragma once
+
+#include <QObject>
+#include <QString>
+#include <QStringList>
+
+#include <unistd.h>
+
+class QSocketNotifier;
+
+class InitiallyStoppedProcess
+{
+public:
+    InitiallyStoppedProcess() = default;
+    ~InitiallyStoppedProcess();
+
+    InitiallyStoppedProcess(const InitiallyStoppedProcess&) = delete;
+    InitiallyStoppedProcess operator=(const InitiallyStoppedProcess&) = delete;
+
+    pid_t processPID() const
+    {
+        return m_pid;
+    }
+
+    bool reset(const QString& exePath, const QStringList& exeOptions, const QString& workingDirectory);
+    bool run();
+    void terminate();
+    void kill();
+
+private:
+    pid_t m_pid = -1;
+};
+
+class PerfControlFifoWrapper : public QObject
```

Comment:
please put these classes into separate files, they are not tightly coupled after all